src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -478,6 +483,9 @@ def load_checkpoint(
         optimzer_state_requested = bool(len([item for item in state.values() if isinstance(item, Optimizer)]))
 
         torch.cuda.empty_cache()
+        with suppress(AttributeError):
+            torch.xpu.empty_cache()
+
```

Comment:
before calling empty_cache, how about detect whether it is cuda or xpu first?

Reply:
In a PR , will merge .